manager/orchestrator/update/updater.go

```diff
@@ -423,6 +428,14 @@ func (u *Updater) updateTask(ctx context.Context, slot orchestrator.Slot, update
 				u.updatedTasksMu.Lock()
 				u.updatedTasks[updated.ID] = time.Now()
 				u.updatedTasksMu.Unlock()
+
+				if startThenStop {
+					_, err := u.store.Batch(func(batch *store.Batch) error {
+						u.removeOldTasks(ctx, batch, slot)
```

Comment:
If `updateTask` returns an error, that error gets logged with the text "update failed"

```
                        if err := u.updateTask(ctx, slot, updated); err != nil {
                                log.G(ctx).WithError(err).WithField("task.id", updated.ID).Error("update failed")
                        }
```

I think it's better to log a warning here, since the update actually happened, though there is a possibility of a task being left over.